src/ui/index.js

```diff
@@ -77,12 +87,74 @@ export default async (kbnServer, server, config) => {
       },
       vars: defaults(app.getInjectedVars() || {}, uiExports.defaultInjectedVars),
     };
-
+    if (kibanaTranslations.length <= 0) {
+      const locale = await getTranslationLocale(acceptLanguages);
+      kibanaTranslations = await i18nPlugin.getRegisteredLocaleTranslations(locale);
+    }
     return this.view(app.templateName, {
       app: app,
       loadingGif: loadingGif,
       kibanaPayload: payload,
       bundlePath: `${config.get('server.basePath')}/bundles`,
+      welcomeMessage: translate('CORE-WELCOME_MESSAGE'),
+      welcomeError: translate('CORE-WELCOME_ERROR'),
     });
   });
 };
+
+function translate(key) {
+  if (!(key in kibanaTranslations)) {
+    return null;
+  }
+  return kibanaTranslations[key];
+}
+
+async function getTranslationLocale(acceptLanguages) {
+  let localeStr = '';
+  let foundLang = false;
+
+  if (acceptLanguages === null || acceptLanguages.length <= 0) {
+    return DEFAULT_LOCALE;
+  }
+
+  const acceptLangsLen = acceptLanguages.length;
+  const registeredLocales = await i18nPlugin.getRegisteredTranslationLocales();
```

Comment:
Sure, changes have now been updated to the i18n plugin. I therefore removed the 'async'.